src/framework/parsers/draco-decoder.js

```diff
@@ -0,0 +1,174 @@
+                        vertices: data.vertices
+                    });
+                }
+                this.jobCallbacks.delete(data.jobId);
+
+                if (this.jobQueue.length > 0) {
+                    const job = this.jobQueue.shift();
+                    this.run(worker, job);
+                } else {
+                    const index2 = this.workers[2].indexOf(worker);
+                    if (index2 !== -1) {
+                        this.workers[2].splice(index2, 1);
+                        this.workers[1].push(worker);
+                    } else {
+                        const index1 = this.workers[1].indexOf(worker);
+                        if (index1 !== -1) {
+                            this.workers[1].splice(index1, 1);
+                            this.workers[0].push(worker);
+                        } else {
+                            // logical error
+                            Debug.error('logical error');
+                        }
+                    }
+                }
+            });
+        });
+    }
+
+    enqueueJob(buffer, callback) {
+        const job = {
+            jobId: this.jobId++,
+            buffer: buffer
+        };
+        this.jobCallbacks.set(job.jobId, callback);
+
+        if (this.workers[0].length > 0) {
+            const worker = this.workers[0].shift();
+            this.workers[1].push(worker);
+            this.run(worker, job);
+        } else if (this.workers[1].length > 0) {
+            const worker = this.workers[1].shift();
+            this.workers[2].push(worker);
+            this.run(worker, job);
+        } else {
+            this.jobQueue.push(job);
+        }
+    }
+}
+
+const defaultNumWorkers = 1;
+
+let jobQueue;
+let lazyConfig;
+
+const initializeWorkers = (config) => {
+    if (jobQueue) {
+        return true;
+    }
+
+    if (!config) {
+        if (lazyConfig) {
+            config = lazyConfig;
+        } else {
+            const moduleConfig = WasmModule.getConfig('DracoDecoderModule');
+            if (moduleConfig) {
+                config = {
+                    jsUrl: moduleConfig.glueUrl,
+                    wasmUrl: moduleConfig.wasmUrl,
+                    numWorkers: moduleConfig.numWorkers
+                };
+            } else {
+                config = {
+                    jsUrl: 'draco.wasm.js',
+                    wasmUrl: 'draco.wasm.wasm',
+                    numWorkers: defaultNumWorkers
+                };
+            }
+        }
+    }
+
+    if (!config.jsUrl || !config.wasmUrl) {
+        return false;
+    }
+
+    // worker urls must be absolute
+    const absoluteUrl = (url) => {
+        return ABSOLUTE_URL.test(url) ? url : new URL(window.location.href).origin + url;
+    };
+
+    const jsUrl = absoluteUrl(config.jsUrl);
+    const wasmUrl = absoluteUrl(config.wasmUrl);
+
+    // create workers
+    const numWorkers = Math.max(1, Math.min(16, config.numWorkers || defaultNumWorkers));
```

Comment:
So if I specified 0 workers, I'd get 1? I _guess_ that's intentional? Just wondering if it should be `??` instead of `||`.

Reply:
yeah right, 0 should become 1